core/condition_test.go

```diff
@@ -274,6 +274,13 @@ func TestCondition_evaluate(t *testing.T) {
 			ExpectedSuccess: true,
 			ExpectedOutput:  "[BODY] == pat(*<div id=\"user\">john.doe</div>*)",
 		},
+		{
+			Name:            "body-pattern-html-failure",
+			Condition:       Condition("[BODY] == pat(*<div id=\"user\">john.doe</div>*)"),
+			Result:          &Result{body: []byte(`<!DOCTYPE html><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8" /></head><body><div id="user">jane.doe</div></body></html>`)},
+			ExpectedSuccess: false,
+			ExpectedOutput:  "[BODY] (<!DOCTYPE html><html lang...(truncated)) == pat(*<div id=\"user\">john.doe</div>*)",
+		},
```

Comment:
Could you add another test case where `pat()` is on the left side of the operator instead of the right side?

e.g. 
```go
        Condition:       Condition("pat(*<div id=\"user\">john.doe</div>*) == [BODY]"),
```